File: cpp/halloweenParty.cpp
// https://www.hackerrank.com/challenges/halloween-party

#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
#include <bits/stdc++.h>
using namespace std;

int main() {
    int T;
    cin >> T;
    assert(1 <= T);
    assert(T <= 10);
    while (T--) {
        long long int N;
        cin >> N;
        assert(1 <= N);
        assert(N <= 10000000);
        if (N % 2 == 0) {
            cout << (N / 2) * (N / 2) << endl;
        } else {
            cout << (N / 2) * (N / 2 + 1) << endl;
        }
    }
    return 0;
}

File: java/src/HalloweenParty.java
// https://www.hackerrank.com/challenges/halloween-party
// t: number of test cases
// T: O(t)
// S: O(1)

import java.util.Scanner;

public class HalloweenParty {
    private static final Scanner SCANNER = new Scanner(System.in);

    public static void main(String[] args) {
        int testCases = SCANNER.nextInt();
        while (testCases-- > 0) {
            final long cuts = SCANNER.nextLong();
            System.out.println(maxBlocks(cuts));
        }
    }

    private static long maxBlocks(final long n) {
        return (n / 2) * (n - n / 2);
    }
}
